### PR TITLE
830 front date range picker

### PR DESCRIPTION
**Pull request by rufataliy**
Fixes #{issue number here}

  - [x] Up to date with `dev` branch
  - [x] Branch name follows [guidelines](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md#feature-branching)
  - [x] All PR Status checks are successful
  - [ ] Peer reviewed and approved

dateselector

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

### CODE REVIEW

File: client/components/DateSelector/options.js
Comment (adamkendis) [inline client/components/DateSelector/options.js:31]: Swap the order of the arguments for `formatDateToMMDD` so the option text shows startdate - enddate instead of the other way around.

File: client/components/DateSelector/options.js
Comment (adamkendis) [inline client/components/DateSelector/options.js:35]: Let's ask UI/UX about the wording of some of these options. It seems weird for the option to say `December` if the true range for this option as of today is `12/21/2020 - 1/21/2021`

File: client/components/common/DatePicker/DatePicker.jsx
Comment (adamkendis) [inline client/components/common/DatePicker/DatePicker.jsx:116]: Update this aria-label to read something like `"toggle calendar datepicker"`

File: client/components/common/DatePicker/DatePicker.jsx
Comment (adamkendis) [inline client/components/common/DatePicker/DatePicker.jsx:91]: Fancy! I've never used this before.

File: client/components/common/customHooks/useOutsideClick.js
Comment (adamkendis) [inline client/components/common/customHooks/useOutsideClick.js:3]: Nice, this is great. Going to add it to the RequestTypeSelector after this is merged in.

**Review by adamkendis — CHANGES_REQUESTED** [review]
Looks excellent! Nice styles on the daypicker. I left some comments on the code and a few other things below.

- I'm getting several proptype errors from the Weekday component.
- If `<DateRanges />` is expanded and the user opens the `<DayPicker />` by clicking the calendar, I think `<DateRanges />` should collapse.
- Lets ask UI/UX about some of the labels in DateRanges.
- We'll need to hook this up to redux but can be done in a different ticket/PR.
- Some fonts need to be updated but no big deal. Can do this when we start adding several selectors to the menu.

File: client/components/DateSelector/options.js
Comment (rufataliy) [reply client/components/DateSelector/options.js:31]: Oh yeah. Good catch.

File: client/components/DateSelector/options.js
Comment (rufataliy) [reply client/components/DateSelector/options.js:35]: Sure. Not sure about the implementation though.

File: client/components/common/DatePicker/DatePicker.jsx
Comment (rufataliy) [reply client/components/common/DatePicker/DatePicker.jsx:116]: Done

File: client/components/common/DatePicker/DatePicker.jsx
Comment (rufataliy) [reply client/components/common/DatePicker/DatePicker.jsx:91]: Oh yeah, It is cleaner.

File: client/components/common/customHooks/useOutsideClick.js
Comment (rufataliy) [reply client/components/common/customHooks/useOutsideClick.js:3]: Maybe we can add it to the selector box as well.

File: client/components/DateSelector/options.js
Comment (rufataliy) [reply client/components/DateSelector/options.js]: > Looks excellent! Nice styles on the daypicker. I left some comments on the code and a few other things below.
> 
> * I'm getting several proptype errors from the Weekday component.
> * If `<DateRanges />` is expanded and the user opens the `<DayPicker />` by clicking the calendar, I think `<DateRanges />` should collapse.
> * Lets ask UI/UX about some of the labels in DateRanges.
> * We'll need to hook this up to redux but can be done in a different ticket/PR.
> * Some fonts need to be updated but no big deal. Can do this when we start adding several selectors to the menu.

- Added date picker close when dropdown opens and vice versa
- Changed the date range options to be consistent with the MMDD MMYYY formats.
- Adjusted font styles 
- Removed array-callback-return and consistent-return rules for linting. NEEDS APPROVAL. It didn't allow to have branched conditional return statement. Check lines 19 and 20 in [this file](https://github.com/hackforla/311-data/blob/830-FRONT-dateRangePicker/client/components/DateSelector/options.js)
- Didn't hook the component to the redux store. We can have another PR for it.

File: client/components/DateSelector/options.js
Comment (adamkendis) [inline client/components/DateSelector/options.js:37]: let's do MMDD here

File: client/components/DateSelector/options.js
Comment (adamkendis) [inline client/components/DateSelector/options.js:44]: and MMDD here

File: client/components/FilterMenu/index.js
Comment (adamkendis) [inline client/components/FilterMenu/index.js:21]: changing strings in this file to use doublequote introduced a bunch of linting errors. I think it's passing the pull request checks because the `--fix` flag was added to the lint script. `npm run lint` before your next push should revert these changes.

File: client/components/common/DatePicker/DatePicker.jsx
Comment (adamkendis) [inline client/components/common/DatePicker/DatePicker.jsx:56]: add whitespace after the `-` to evenly space `from - to`
```suggestion
        -
        {' '}
```

File: client/components/common/ReactDayPicker/Weekday.jsx
Comment (adamkendis) [inline client/components/common/ReactDayPicker/Weekday.jsx:2]: nice! looks like you figured out the props errors.

**Review by adamkendis — CHANGES_REQUESTED** [review]
hey @rufataliy , changes look good but there's still a problem with how the dates are being displayed. I left a few suggestions to fix that problem and a couple tiny formatting things I missed before.

File: client/components/DateSelector/options.js
Comment (adamkendis) [reply client/components/DateSelector/options.js]: `Removed array-callback-return and consistent-return rules for linting. NEEDS APPROVAL. It didn't allow to have branched conditional return statement. Check lines 19 and 20 in this file`

left comments in another review below

File: client/components/DateSelector/options.js
Comment (rufataliy) [reply client/components/DateSelector/options.js:18]: Good catch on that.

File: client/components/DateSelector/options.js
Comment (rufataliy) [reply client/components/DateSelector/options.js:44]: What about the last six months spans back to the last year? For example from January 2021 to July 2020.

File: client/components/common/ReactDayPicker/Weekday.jsx
Comment (rufataliy) [reply client/components/common/ReactDayPicker/Weekday.jsx:2]: Yeah. I got it working.

File: client/.eslintrc.js
Comment (rufataliy) [reply client/.eslintrc.js:34]: Sure. Thanks

File: client/components/FilterMenu/index.js
Comment (rufataliy) [reply client/components/FilterMenu/index.js:21]: Yeah. I think we should have linitng as pre commit tasks for git.

File: client/components/DateSelector/options.js
Comment (rufataliy) [reply client/components/DateSelector/options.js:20]: I just added a guard at the beginning of the function and return undefined in the arrow function. It is more robust.

File: client/components/FilterMenu/index.js
Comment (adamkendis) [reply client/components/FilterMenu/index.js:21]: Sounds good to me, will get that pre commit hook in.

File: client/components/DateSelector/options.js
Comment (adamkendis) [reply client/components/DateSelector/options.js:44]: Yeah, considered that as well. I figured for any timespan less than 1 year, it's useful and more informative for the user to show DD in the displayed date range. What do you think?

File: client/components/FilterMenu/index.js
Comment (rufataliy) [reply client/components/FilterMenu/index.js:21]: I already have a ticket for it. https://github.com/hackforla/311-data/projects/10#card-53425622

File: client/components/DateSelector/options.js
Comment (adamkendis) [reply client/components/DateSelector/options.js:44]: Per team discussion, remove displayed date ranges from dropdown.

**Review by adamkendis — APPROVED** [review]
looks good to me!

**Pull request merged**
